### ISSUE TITLE
IParameterTypeDefinition should have parameter "transformer" set to optional

### ISSUE THREAD

**Issue by hammzj** [badeball/cypress-cucumber-preprocessor#1179]
### Current behavior

https://github.com/badeball/cypress-cucumber-preprocessor/blob/055d8df6a62009c94057b0d894a30e142cb87b94/lib/public-member-types.ts#L3-L7

`transformer` is required, but should be optional. If a transformer is not provided for a parameter type that returns a single capture group, then it will return the original value as a string.

However, if a parameter type has two capture groups, it needs a transformer that returns all arguments, or else it only returns only the first found argument in a capture group. That, however, does not require a transformer as the code will still execute.

### Desired behavior

Make `transformer` be optional:

```ts
export interface IParameterTypeDefinition<T, C extends Mocha.Context> {
  name: string;
  regexp: RegExp;
  transformer?: (this: C, ...match: string[]) => T;
}

```

### Test code to reproduce

```ts
defineParameterType({
	name: "threeChoices",
	regexp: /(now|later|maybe)/
});

defineParameterType({
	name: "someNumber",
	regexp: /(\d+)/
});


Given('call me {threeChoices}', function(choice) {
  cy.log(choice, typeof choice) // => string
});


Given('number of {someNumber}', function(x) {
  cy.log(x, typeof x) // => string
});
```

```cucumber
Feature: Demo

  Scenario Outline: Test
    * call me <choice>
    * number of 2
    * number of 3.14
    Examples:
      | choice |
      | now    |
      | later  |
      | maybe  |
```

### Versions

* **Cypress version**: 13.1
* **Preprocessor version**: 18.0.6
* **Node version**: 18.15.0

### Checklist

- [x] I've read the [FAQ](https://github.com/badeball/cypress-cucumber-preprocessor/blob/master/docs/faq.md).
- [x] I've read [instructions for logging issues](https://github.com/badeball/cypress-cucumber-preprocessor/blob/master/CONTRIBUTING.md#bug-reports).
- [x] I'm not using `cypress-cucumber-preprocessor@4.3.1` (package name has changed and it is no longer the most recent version, see [#689](https://github.com/badeball/cypress-cucumber-preprocessor/issues/689)).

**Comment by hammzj** [comment]
See constructor: https://github.com/cucumber/cucumber-expressions/blob/b4d5d7025ecfe7405b5235bade7acce1d827b30b/javascript/src/ParameterType.ts#L45-L62

This also supports `type` to directly define the type, but not sure if needed.

**Comment by badeball** [comment]
Hi @hammzj, I've fixed with this v20.0.4.
